### ISSUE TITLE
Should we use eval for discriminator in the training of generator if dropout is used?

### ISSUE THREAD

**Issue by kcltfask** [junyanz/pytorch-CycleGAN-and-pix2pix#902]
In training of generator, if dropout is used in discriminator, the GAN loss for generator calculated will be affected by the dropout effect of discriminator.  

        # GAN loss D_A(G_A(A))
        self.loss_G_A = self.criterionGAN(self.netD_A(self.fake_B), True)
        # GAN loss D_B(G_B(B))
        self.loss_G_B = self.criterionGAN(self.netD_B(self.fake_A), True)

So should we use netD_A.eval() and netD_B.eval() before the calculation of GAN loss in order to eliminate the dropout effect?

**Comment by junyanz** [comment]
The Dropout probably will not affect the training of the generator a lot. But feel free to experiment with it.
